### ISSUE TITLE
Angular universal: Should I init the use of the lang on both sides, client and server?

### ISSUE THREAD

**Issue by peterpeterparker** [ngx-translate/core#625]
**I'm submitting a ...**  (check one with "x")
[X] support request => check the FAQ and search github for a similar issue before submitting

**Current behavior**

I'm really not sure about the initialization of ngx-translate when integrating it with angular universal. should I initialize the language in both side, client and server, or only on the server side?

Respectively in `app.component.ts` right now I do like following:

    ngOnInit() {
   
        this.translateService.setDefaultLang('en');

        if (isPlatformServer(this.platformId)) {
              let acceptLanguage: string = this.req.headers['accept-language'];
              let languages: string[] = acceptLanguage.match(/[a-zA-Z\-]{2,10}/g) || [];
        
              if (languages.length > 0) {
                let userLang: string = languages[0].split('-')[0];
                userLang = /(de|en|fr|it)/gi.test(userLang) ? userLang : 'en';
        
                this.translateService.use(userLang);
              } else {
                this.translateService.use('en');
              }
            } else if (isPlatformBrowser(this.platformId)) { // <-- Here, should I do that part?
              let userLang = navigator.language.split('-')[0];
              userLang = /(de|en|fr|it)/gi.test(userLang) ? userLang : 'en';
        
              this.translateService.use(userLang);
            }
        }

But doing so, when I do a full refresh of my page I see a really fast refresh of my labels, is that correct?

Also if I don't initialize it on the client side (`isPlatformBrowser`) it look like it isn't working because first I see the right translation before seeing my key name instead of the real text.

Thx in advance for the advice, sorry slowly a bit lost and slow to get it

**Comment by peterpeterparker** [comment]
I had a look to the app of @premiumwd  where both ngx-translate and angular universal and it looks like that he's initializing ngx-translate on both side

https://github.com/premiumwd/d/blob/master/src/app/app.component.ts

**Comment by peterpeterparker** [comment]
After much thinking about it I think that ngx-translate should be initialized on both side for the simple reason that this let me offer users a dynamic language change on the client side without having to query the server to rebuild the page

**Comment by peterpeterparker** [comment]
for the record, after much much thinking, I should right now only load the translations on the server side, see https://github.com/ngx-translate/core/issues/754

**Comment by ocombe** [comment]
Yes I was more thinking about a custom loader server side that stores in the state transfer service, and a custom loader client side that pulls from the state transfer service, or http if the state service is empty for this lang

**Comment by peterpeterparker** [comment]
@ocombe you are definitely right, it's way more clean to use custom loaders on both side with the help of transfer-state and http as fallback than my first solution. I have now implemented it and even deployed it in production and it still works like a charm!

To reflect this, I have updated all my posts, solution and shared the code of the loaders in https://github.com/ngx-translate/core/issues/754

thx for your amazing help and support. 

Have a merry Christmas 🎄
